primedev/windows/libsys.cpp

```diff
@@ -0,0 +1,122 @@
+#include "libsys.h"
+#include "plugins/pluginmanager.h"
+
+#define XINPUT1_3_DLL "XInput1_3.dll"
+
+typedef HMODULE (*WINAPI ILoadLibraryA)(LPCSTR lpLibFileName);
+typedef HMODULE (*WINAPI ILoadLibraryExA)(LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+typedef HMODULE (*WINAPI ILoadLibraryW)(LPCWSTR lpLibFileName);
+typedef HMODULE (*WINAPI ILoadLibraryExW)(LPCWSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+
+ILoadLibraryA o_LoadLibraryA = nullptr;
+ILoadLibraryExA o_LoadLibraryExA = nullptr;
+ILoadLibraryW o_LoadLibraryW = nullptr;
+ILoadLibraryExW o_LoadLibraryExW = nullptr;
+
+//-----------------------------------------------------------------------------
+// Purpose: Run detour callbacks for given HMODULE
+//-----------------------------------------------------------------------------
+void LibSys_RunModuleCallbacks(HMODULE hModule)
+{
+	if (!hModule)
+	{
+		return;
+	}
+
+	// Get module base name in ASCII as noone wants to deal with unicode
+	CHAR szModuleName[MAX_PATH];
+	GetModuleBaseNameA(GetCurrentProcess(), hModule, szModuleName, MAX_PATH);
+
+	// DevMsg(eLog::NONE, "%s\n", szModuleName);
+
+	// Call callbacks
+	CallLoadLibraryACallbacks(szModuleName, hModule);
+	g_pPluginManager->InformDllLoad(hModule, fs::path(szModuleName));
+}
+
+//-----------------------------------------------------------------------------
+// Load library callbacks
+
+HMODULE WINAPI WLoadLibraryA(LPCSTR lpLibFileName)
+{
+	HMODULE hModule = o_LoadLibraryA(lpLibFileName);
+
+	LibSys_RunModuleCallbacks(hModule);
+
+	return hModule;
+}
+
+HMODULE WINAPI WLoadLibraryExA(LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags)
+{
+	HMODULE hModule;
+
+	LPCSTR lpLibFileNameEnd = lpLibFileName + strlen(lpLibFileName);
+	LPCSTR lpLibName = lpLibFileNameEnd - strlen(XINPUT1_3_DLL);
+
+	// replace xinput dll with one that has ASLR
+	if (lpLibFileName <= lpLibName && !strncmp(lpLibName, XINPUT1_3_DLL, strlen(XINPUT1_3_DLL) + 1))
+	{
+		hModule = o_LoadLibraryExA("XInput9_1_0.dll", hFile, dwFlags);
+
+		if (!hModule)
+		{
+			spdlog::error("Could not find XInput9_1_0.dll");
+
+			return nullptr;
+		}
```

Comment:
We can't handle it like this. it causes the game to quietly close with only a single error log in northstar

There's two ways we can work around this issue:
1. Restore the previous behaviour, where a message box is shown to the user
2. The following, which allows the game to run with no controller input:
```suggestion
		if (!hModule)
		{
			spdlog::error("Could not find XInput9_1_0.dll");

			// return -1 so that the game doesn't close (it has a check for if the hModule is 0 which then closes the game)
			return (HMODULE)(-1);
		}
```